QEfficient/generation/llm_generator.py

```diff
@@ -0,0 +1,298 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c)  2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+import json
+import os
+import torch
+import numpy as np
+import torch.nn as nn
+
+from typing import Dict, List, Optional, Union
+from threading import Thread
+
+from QEfficient.generation.cloud_infer import QAICInferenceSession
+
+from transformers import (
+    AutoTokenizer,
+    TextIteratorStreamer,
+    TextStreamer,
+    AutoTokenizer,
+    LogitsProcessorList,
+    MinLengthLogitsProcessor,
+    TopKLogitsWarper,
+    TemperatureLogitsWarper,
+    StoppingCriteriaList,
+    MaxLengthCriteria,
+)
+
+
+
+class LLMGenerator:
+    def __init__(
+        self,
+        qpc_path,
+        model_name,
+        device_id: Optional[List[int]] = [0],
+        prompt_len: Optional[int] = 32,
+        ctx_len: Optional[int] = 128,
+        streamer: Optional[Union[TextStreamer, TextIteratorStreamer]] = None,
+        logits_processor: Optional = None,
+        logits_warper: Optional = None,
+    ):
+        self.session = None
+        self.tokenizer = None
+        self.is_first_prompt = False
+        self.model_name = ""
```

Comment:
Can be replaced with line 66, is there a purpose of assigning empty string here?